include/zephyr/kernel.h

```diff
@@ -205,6 +205,13 @@ extern void k_thread_foreach_unlocked(
  */
 #define K_CALLBACK_STATE (BIT(4))
 
+#if defined(CONFIG_DSP)
+/**
+ * @brief DSP registers are managed by context switch
+ */
+#define K_DSP_REGS (BIT(5))
```

Comment:
I don't see any code (not even test code) that's setting/clearing this bit. Shouldn't there be some?

Reply:
Added DSP_SHARING disable & enable API functions.